tests/bsim/bluetooth/host/scan/start_stop/src/main.c

```diff
@@ -0,0 +1,236 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/irq.h>
+#include <string.h>
+
+#include <zephyr/logging/log.h>
+#include <zephyr/settings/settings.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/atomic_builtin.h>
+#include <babblekit/testcase.h>
+#include <babblekit/sync.h>
+
+LOG_MODULE_REGISTER(bt_bsim_scan_start_stop, LOG_LEVEL_DBG);
+
+#define WAIT_TIME_S 60
+#define WAIT_TIME   (WAIT_TIME_S * 1e6)
+
+static atomic_t flag_adv_report_received;
+static atomic_t flag_periodic_sync_established;
+static bt_addr_le_t adv_addr;
+
+static void test_tick(bs_time_t HW_device_time)
+{
+	if (bst_result != Passed) {
+		TEST_FAIL("Test failed (not passed after %d seconds)\n", WAIT_TIME_S);
+	}
+}
+
+static void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(WAIT_TIME);
+}
+
+static void bt_sync_established_cb(struct bt_le_per_adv_sync *sync,
+			    struct bt_le_per_adv_sync_synced_info *info)
+{
+	LOG_DBG("Periodic sync established");
+	atomic_set(&flag_periodic_sync_established, true);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = bt_sync_established_cb,
+};
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	memcpy(&adv_addr, addr, sizeof(adv_addr));
+
+	bt_addr_le_to_str(&adv_addr, addr_str, sizeof(addr_str));
+	LOG_DBG("Device found: %s (RSSI %d), type %u, AD data len %u",
+	       addr_str, rssi, type, ad->len);
+	atomic_set(&flag_adv_report_received, true);
+}
+
+void run_dut(void)
+{
+	LOG_DBG("start");
```

Comment:
Can you add a description of this test?
E.g. What's its main purpose, and how it achieves it.
something like this:
https://github.com/zephyrproject-rtos/zephyr/blob/2ecdb6e53250ad5547e12918f68cb6f5a740668f/tests/bsim/bluetooth/host/misc/sample_test/src/dut.c#L67-L88